File: hw/chicaiza/u1/hw9AntSimulator/AntSimulator/src/model/Colony.java
package model;

import java.util.List;

public class Colony {
    private Nest nest;
    private List<Ant> ants;
    
    public Colony(Nest nest) {
        this.nest = nest;
        // Initialize the ant list if needed
    }

    // Get the colony's nest
    public Nest getNest() {
        return nest;
    }

    // Add an ant to the colony
    public void addAnt(Ant ant) {}

    // Update all ants in this colony for a tick
    public void runTick() {}
}

File: hw/bonilla/u1/HW09AntSimulator/Colony.js
export class Colony {
  constructor(nest) {
    this.nest = nest;
    this.ants = new Set();
  }

  addAnt(ant) {
    this.ants.add(ant);
    ant.colony = this;
  }

  removeAnt(ant) {
    this.ants.delete(ant);
  }

  tick(simulation) {
    while (this.nest.canCreateAnt()) {
      this.nest.createAnt(this);
    }
  }
}